subsys/net/lib/lwm2m/Kconfig

```diff
@@ -56,6 +56,16 @@ config LWM2M_ENGINE_MESSAGE_HEADER_SIZE
 	help
 	  Extra room allocated to handle CoAP header data
 
+config LWM2M_COAP_MAX_MSG_SIZE
+	int "LWM2M CoAP maximum message size"
+	default LWM2M_COAP_BLOCK_SIZE
+	range 64 8168
+	help
+	  CoAP message size used by LWM2M. Minimum is the block size used
+	  in blockwise transfers. Packet size(msg plus header) must not exceed 8kB.
+	  The 8kB upper limit is arbitrary. Theoretical max would be:
```

Comment:
I'm not sure, do we need to impose any arbitrary limit on LwM2M layer? Assuming IP-level fragmentation is supported, it should be fine to support UDP limits.

And note, that IP fragmentation is disabled in Zephyr by default and **it's only supported by IPv6**. Therefore I think it should also be worth to mention that the user have to make sure that the network configuration he uses needs to have IP fragmentation support enabled it this value is near or above network MTU.

Reply:
Good point. I'll set the upper limit to be the same as the theoretical one. There would be need to extract the size of the UDP, IPv4 and IPv6 headers in use but maybe that's a moving target we don't consider in this case.

Reply:
Maybe we could have this as big as needed, and then have it in a separate task to implement splitting larger than PMTU packets into CoAP block-wise requests.
Because even if we agree to split messages, we still need at lest one big message buffer where we would start the splitting.

Also, lets not try to guess the underlying transport protocol. It might be UDP or DTLS, in which case we should handle the PMTU limits (to avoid fragmentation), but it could also be stream, like TCP or TLS, in which case fragmentation does not happen, or matter.